src/resources/2017/Spacegate.ts

```diff
@@ -0,0 +1,155 @@
+import { cliExecute, Item, visitUrl } from "kolmafia";
+import { get } from "../../property";
+import { $item, $items } from "../../template-string";
+
+export function have(): boolean {
+  return get("spacegateAlways") || get("_spacegateToday");
+}
+
+export function checkStatus(): void {
+  visitUrl("place.php?whichplace=spacegate&action=sg_Terminal");
+}
+
+export function dialled(): boolean {
+  checkStatus();
+  return get("_spacegateCoordinates") !== "";
+}
+
+export function hazards() {
+  return get("_spacegateHazards");
+}
+
+export function planetName() {
+  return get("_spacegatePlanetName");
+}
+
+export function planetCoords() {
+  return get("_spacegateCoordinates");
+}
+
+export function plantLife() {
+  return get("_spacegatePlantLife");
+}
+
+export function animalLife() {
+  return get("_spacegateAnimalLife");
+}
+
+export function intelligentLife() {
+  return get("_spacegateIntelligentLife");
+}
+
+export function ruins() {
+  return get("_spacegateRuins");
+}
+
+export function spants() {
+  return get("_spacegateSpant");
+}
+
+export function murderBots() {
+  return get("_spacegateMurderbot");
+}
+
+export function hazardEquipment(hazards: string): Item | Item[] {
+  const equipment: Item[] = $items`none`;
+  if (hazards.includes("toxic atmosphere"))
+    equipment.push($item`filter helmet`);
+  if (hazards.includes("high gravity"))
+    equipment.push($item`exo-servo leg braces`);
+  if (hazards.includes("irradiated")) equipment.push($item`rad cloak`);
+  if (hazards.includes("magnetic storms"))
+    equipment.push($item`gate transceiver`);
+  if (hazards.includes("high winds"))
+    equipment.push($item`high-friction boots`);
+  if (equipment.length === 2) {
+    return equipment[1];
+  } else {
+    return equipment;
+  }
```

Comment:
```suggestion
const hazardEquipment = {
  "toxic atmosphere": $item`filter helmet`,
  "high gravity": $item`exo-servo leg braces`,
  "irradiated": $item`rad cloak`,
  "magnetic storms": $item`gate transceiver`,
  "high winds": $item`high-friction boots`,
};
return Object.entries(hazardEquipment).filter(([cue]) => hazard.includes(clue)).map(([, item]) => item)
```
or something like that?

As I write that, that specifically doesn't look great. but packing all of that information into a Map or an Object or something feels like a much more natural approach.